app/code/Magento/Catalog/Model/Product/Compare/ListCompare.php

```diff
@@ -51,6 +51,13 @@ class ListCompare extends \Magento\Framework\DataObject
      */
     protected $_compareItemFactory;
 
+    /**
+     * Catalog factory
+     *
+     * @var \Magento\Catalog\Model\ProductFactory
+     */
+    protected $_catalogFactory;
```

Comment:
Please add new properties with the `private` scope of visibility.
I would also recommend avoiding underscores at the beginning of the private/protected method's name since this is the legacy approach according to the PSR-2 Coding Style Guide:
```
Property names SHOULD NOT be prefixed with a single underscore to indicate protected or private visibility.
```

Reply:
changed to private and removed underscore.